src/Microsoft.AspNet.Mvc.Core/ModelBinding/MutableObjectModelBinder.cs

```diff
@@ -485,26 +471,6 @@ internal static PropertyValidationInfo GetPropertyValidationInfo(ModelBindingCon
                 value = dtoResult.Model;
             }
 
-            // 'Required' validators need to run first so that we can provide useful error messages if
-            // the property setters throw, e.g. if we're setting entity keys to null.
```

Comment:
> With this change, the first error message could be from an Exception and not the required validator.

That's exactly the point. The model property is only set when `isModelSet == true`. An exception can only occur in a case where the model binder has returned a value that can't be set **and** indicated that it should be set on the model.